plugins/inputs/cisco_telemetry_mdt/cisco_telemetry_mdt.go

```diff
@@ -50,13 +52,974 @@ type CiscoTelemetryMDT struct {
+		return str
+	case *telemetry.TelemetryField_Sint32Value:
+		str = strconv.FormatInt(int64(val.Sint32Value), 10)
+		return str
+	case *telemetry.TelemetryField_Sint64Value:
+		str = strconv.FormatInt(val.Sint64Value, 10)
+		return str
+	}
+	return ""
+}
+
+//xform Uint64 to int64
+func (c *CiscoTelemetryMDT) nxosValueXformUint64Toint64(field *telemetry.TelemetryField, value interface{}) interface{} {
+	if field.GetUint64Value() != 0 {
+		return int64(value.(uint64))
+	}
+	return nil
+}
+
+//xform string to float
+func (c *CiscoTelemetryMDT) nxosValueXformStringTofloat(field *telemetry.TelemetryField, value interface{}) interface{} {
+	//convert property to float from string.
+	vals := field.GetStringValue()
+	if vals != "" {
+		if valf, err := strconv.ParseFloat(vals, 64); err == nil {
+			return valf
+		}
+	}
+	return nil
+}
+
+//xform string to uint64
+func (c *CiscoTelemetryMDT) nxosValueXformStringToUint64(field *telemetry.TelemetryField, value interface{}) interface{} {
+	//string to uint64
+	vals := field.GetStringValue()
+	if vals != "" {
+		if val64, err := strconv.ParseUint(vals, 10, 64); err == nil {
+			return val64
+		}
+	}
+	return nil
+}
+
+//xform string to int64
+func (c *CiscoTelemetryMDT) nxosValueXformStringToInt64(field *telemetry.TelemetryField, value interface{}) interface{} {
+	//string to int64
+	vals := field.GetStringValue()
+	if vals != "" {
+		if val64, err := strconv.ParseInt(vals, 10, 64); err == nil {
+			return val64
+		}
+	}
+	return nil
+}
+
+//auto-xform
+func (c *CiscoTelemetryMDT) nxosValueAutoXform(field *telemetry.TelemetryField, value interface{}) interface{} {
+	//check if we want auto xformation
+	vals := field.GetStringValue()
+	if vals != "" {
+		if val64, err := strconv.ParseUint(vals, 10, 64); err == nil {
+			return val64
+		}
+		if valf, err := strconv.ParseFloat(vals, 64); err == nil {
+			return valf
+		}
+		if val64, err := strconv.ParseInt(vals, 10, 64); err == nil {
+			return val64
+		}
+	} // switch
+	return nil
+}
+
+//auto-xform float properties
+func (c *CiscoTelemetryMDT) nxosValueAutoXformFloatProp(field *telemetry.TelemetryField, value interface{}) interface{} {
+	//check if we want auto xformation
+	vals := field.GetStringValue()
+	if vals != "" {
+		if valf, err := strconv.ParseFloat(vals, 64); err == nil {
+			return valf
+		}
+	} // switch
+	return nil
+}
+
+//xform uint64 to string
+func (c *CiscoTelemetryMDT) nxosValueXformUint64ToString(field *telemetry.TelemetryField, value interface{}) interface{} {
+	switch val := field.ValueByType.(type) {
+	case *telemetry.TelemetryField_StringValue:
+		if len(val.StringValue) > 0 {
+			return val.StringValue
+		}
+	case *telemetry.TelemetryField_Uint64Value:
+		return (strconv.FormatUint(val.Uint64Value, 10))
+	}
+	return nil
+}
+
+//Xform value field.
+func (c *CiscoTelemetryMDT) nxosValueXform(field *telemetry.TelemetryField, value interface{}, path string) interface{} {
+	//if isNXOS
+	if !strings.ContainsRune(path, ':') {
```

Comment:
Please invert the logic to safe some indention levels
```suggestion
	if strings.ContainsRune(path, ':') {
		// not NXOS
		return nil
	}
```